crates/stackable-operator/src/builder/pod/mod.rs

```diff
@@ -254,28 +269,51 @@ impl PodBuilder {
         self
     }
 
-    pub fn add_volume(&mut self, volume: Volume) -> &mut Self {
-        self.volumes.get_or_insert_with(Vec::new).push(volume);
-        self
-    }
-
     /// Utility function for the common case of adding an emptyDir Volume
     /// with the given name and no medium and no quantity.
     pub fn add_empty_dir_volume(
         &mut self,
         name: impl Into<String>,
         quantity: Option<Quantity>,
-    ) -> &mut Self {
+    ) -> Result<&mut Self> {
         self.add_volume(
             VolumeBuilder::new(name)
                 .with_empty_dir(None::<String>, quantity)
                 .build(),
         )
     }
 
-    pub fn add_volumes(&mut self, volumes: Vec<Volume>) -> &mut Self {
-        self.volumes.get_or_insert_with(Vec::new).extend(volumes);
-        self
+    /// This function only adds the [`Volume`] in case there is no volume with the same name already.
+    /// In case there already was a volume with the same name already, an [`tracing::error`] is raised in case the
+    /// contents of the volumes differ.
+    ///
+    /// Historically this function unconditionally added volumes, which resulted in invalid [`PodSpec`]s, as volumes
+    /// where added multiple times - think of Trino using the same [`crate::commons::s3::S3Connection`] two times,
+    /// resulting in e.g. the s3 credentials being mounted twice as the sake volume.
```

Comment:
suggestion: Rewording

```suggestion
    /// Adds a new [`Volume`] to the pod while ensuring that no duplicate exists.
    ///
    /// This function ensures that no two [`Volume`]s with the same mname can be
    /// added. An appropriate error is returned when such a duplicate name is
    /// encountered.
    ///
    /// ### Note
    ///
    /// Previously, this function unconditionally added [`Volume`]s, which
    /// resulted in invalid [`PodSpec`]s.
```

Reply:
We only have access to the volume, not the volumeMount, so we don't know the mountPath.
We could only say 'the volume "tls" clashes, bruh', which IMHO is too less information. It's not a strong opinion but I think we would be very happy to know what that volume is (emptyDir, configMap or secret-op volume) in case we get such errors.

Reply:
I'll wait on https://github.com/stackabletech/operator-rs/pull/871#discussion_r1771122689 before changing things

Reply:
Implemented in a sync meeting